doc/website/blog/2022-03-14-ent-hooks-examples.md

```diff
@@ -0,0 +1,430 @@
+---
+title: A few examples for Ent hooks  
+author: Yoni Davidson
+authorURL: "https://github.com/yonidavidson"
+authorImageURL: "https://avatars0.githubusercontent.com/u/5472778"
+authorTwitter: yonidavidson
+---
+Hooks are a popular subject in our blog posts
+[Building Observable Ent Applications with Prometheus](/blog/2021/08/12/building-observable-ent-application-with-prometheus)
+[Sync Changes to External Data Systems using Ent Hooks](/blog/2021/11/1/sync-to-external-data-systems-using-hooks)
+but their reactive nature makes them a tool that is usually affiliated with more advanced users.
+My goal in this blog post is to show you a few use cases where hooks can help you when building servers, and provide you with a few go-to recipes.
+These examples can be later reviewed in this [repo](https://github.com/yonidavidson/ent-hooks-examples).
+
+### So first, let us review in a few words what are hooks?
+
+[Hooks](https://entgo.io/docs/hooks) are a feature of Ent that allows adding custom logic before and after operations that change the data entities.
+
+A mutation is an operation that creates a change in the database.
+There are 5 types of mutations:
+1. Create.
+2. UpdateOne.
+3. Update.
+4. DeleteOne.
+5. Delete.
+
+Hooks are functions that get and return an [ent.Mutator](https://pkg.go.dev/entgo.io/ent#Mutator).
+They function similar to the popular [HTTP middleware pattern](https://github.com/go-chi/chi#middleware-handlers).
```

Comment:
chi is cool but it didn't invent middleware, better reference a more canonical reference such as Wikipedia or an academic paper or a book excerpt.